lib/open_food_network/sales_tax_report.rb

```diff
@@ -24,19 +31,39 @@ def orders
     end
 
     def table
-      orders.map do |order|
-        totals = totals_of order.line_items
-        shipping_cost = shipping_cost_for order
-
-        [order.number, order.created_at, totals[:items], totals[:items_total],
-         totals[:taxable_total], totals[:sales_tax], shipping_cost, order.shipping_tax, order.enterprise_fee_tax, order.total_tax,
-         order.bill_address.full_name, order.distributor.andand.name]
+      case params[:report_type]
+      when "tax_types"
+        orders.map do |order|
+          totals = totals_of order.line_items
+          shipping_cost = shipping_cost_for order
+
+          [order.number, order.created_at, totals[:items], totals[:items_total],
+           totals[:taxable_total], totals[:sales_tax], shipping_cost, order.shipping_tax, order.enterprise_fee_tax, order.total_tax,
+           order.bill_address.full_name, order.distributor.andand.name]
+        end
+      else
+        orders.map do |order|
+          [order.number, order.total - order.total_tax] +
+            relevant_rates.map { |rate| order.tax_adjustment_totals.fetch(rate, 0) } +
+            [order.total_tax, order.display_total]
+        end
       end
+
     end
 
 
     private
 
+    def relevant_rates
+      queries = [ search.result.joins(:line_items => {:adjustments => :tax_rate}).select('spree_tax_rates.*').uniq,
+                  search.result.joins(:adjustments => :tax_rate).select('spree_tax_rates.*').uniq ]
+      queries.map do |query|
+        ActiveRecord::Base.connection.select_all(query)
+      end.sum.map do
```

Comment:
Maybe consider splitting some of this out into a separate method to improve readability/testability?